dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -882,6 +902,266 @@ private void updateInfo(Workflow oldWorkflow, Workflow newWorkflow) {
 
     }
 
+    private void fillInMetadata(DepositMetadata depositMetadata, Workflow workflow, WorkflowVersion workflowVersion) {
+        // add some metadata to the deposition that will be published to Zenodo
+        depositMetadata.setTitle(workflow.getWorkflowPath());
+        // The Zenodo deposit type for Dockstore will always be SOFTWARE
+        depositMetadata.setUploadType(DepositMetadata.UploadTypeEnum.SOFTWARE);
+        // A metadata description is required for Zenodo
+        String description = workflow.getDescription();
+        String descriptionStr = (description == null || description.isEmpty()) ? "n/a" : workflow.getDescription();
+        depositMetadata.setDescription(descriptionStr);
+        depositMetadata.setVersion(workflowVersion.getName());
+
+        // Use the Dockstore workflow labels as Zenodo free form keywords for this deposition.
+        Set<Label> workflowLabels = workflow.getLabels();
+        Iterator labelIter = workflowLabels.iterator();
+        List<String> labelList = new ArrayList<>();
+        while (labelIter.hasNext()) {
+            String label = ((Label)labelIter.next()).getValue();
+            labelList.add(label);
+        }
+        depositMetadata.setKeywords(labelList);
+
+
+        // Get the aliases for this workflow and add them to the deposit
+        // The alias must be a format supported by Zenodo such as
+        // DOI, Handle, ARK...URNs and URLs
+        // See http://developers.zenodo.org/#representation 'related identifiers'
+        Set<String> workflowAliases = workflow.getAliases().keySet();
+        //Set<String> workflowAliases = new HashSet<String>();
+        //String fullWorkflowPath = workflow.getWorkflowPath();
+        //workflowAliases.add("https://dockstore.org/workflows/" + fullWorkflowPath + ":" + workflowVersion.getName());
+
+        Iterator iter = workflowAliases.iterator();
+        List<RelatedIdentifier> aliasList = new ArrayList<RelatedIdentifier>();
+        while (iter.hasNext()) {
+            RelatedIdentifier relatedIdentifier = new RelatedIdentifier();
+            String workflowAlias = ((String)iter.next());
+            relatedIdentifier.setIdentifier(workflowAlias);
+            relatedIdentifier.setRelation(RelatedIdentifier.RelationEnum.ISIDENTICALTO);
+            aliasList.add(relatedIdentifier);
+        }
+        depositMetadata.setRelatedIdentifiers(aliasList);
+
+
+        String wfAuthor = workflow.getAuthor();
+        String authorStr = (wfAuthor == null || wfAuthor.isEmpty()) ? "n/a" : workflow.getAuthor();
+        Author author = new Author();
+        author.setName(authorStr);
+        depositMetadata.setCreators(Arrays.asList(author));
+
+        // A communities entry must not be null, but it can be a null
+        // List for Zenodo
+        List<Community> communities = depositMetadata.getCommunities();
```

Comment:
Maybe we should just make a default community for new zenodo entries, can the end user move entries between communities after the entry is created?

Reply:
The user can remove a workflow from a community if they are a curator, and a user can request that a workflow be added to a community where they are not a curator.

Reply:
Hmmm, so if I'm reading this correctly. Regular non-curator users in a Zenodo "dockstore community" wouldn't be able to move their workflows out to their own community even if they own their workflow (since it was created using their credentials).

Reply:
I create an upload to Zenodo using my credentials. I can edit my workflow deposition metadata on the Zenodo UI, which allows me to delete community information from that metadata. When adding my workflow to a community I believe (web site is giving me a server error now) it sends a request to the community curator to add the workflow.